### DISCUSSION TITLE
Lesson 13: upgradeTo is deprecated .. again

### DISCUSSION THREAD

**Discussion by sunnyStefi** [Cyfrin/foundry-full-course-cu#1033] [Ideas]
Hei coders!

It's super awesome that openzeppelin is updating their contracts every few months 😮 ! But this gives us always the downgrading version problem, **that's not really a solution, isn't it?** 🙈 🙉 🙊 We already moved from `version 4` to `version 5` !

Since we are using just few libs, it would be awesome to have the course coded checked and updated **(at least on github)** whenever openzeppelin or others release a new version.

ref [Openzeppelin docs](https://docs.openzeppelin.com/contracts/5.x/backwards-compatibility#major_releases)

### Major Releases
_Major releases should be assumed incompatible. Nevertheless, the external interfaces of contracts will remain compatible if they are standardized, or if the maintainers judge that changing them would cause significant strain on the ecosystem.
An important aspect that major releases may break is "upgrade compatibility", in particular storage layout compatibility. It will never be safe for a live contract to upgrade from one major release to another._

This happened already with the ERC20Mock constructor, now with the `upgradeTo` in place of `upgradeToAndCall` inside `UUPSUpgradeable.sol`.

It looks like a **recurrent** **problem**. We, as community can help for sure with a push but maybe it would be more helpful just to have an actualised code on github?

**Comment by cromewar** [comment]
Amazing @sunnyStefi, can you send a PR to the Chronological updates?

**Reply by sunnyStefi** [reply]
Done!
